services/graph/pkg/service/v0/utils.go

```diff
@@ -454,3 +458,19 @@ func roleConditionForResourceType(ri *storageprovider.ResourceInfo) (string, err
 		return "", errorcode.New(errorcode.InvalidRequest, "unsupported resource type")
 	}
 }
+
+// GetShareID returns a valid share ID from the given resource ID.
+// Basically, it just checks if the given ID has an opaque ID that is
+// compatible with the id format defined by storagespace package.
+func GetShareID(shareID storageprovider.ResourceId) (*collaboration.ShareId, error) {
```

Comment:
Hm, the naming of the shareID parameter seems wrong. It's not a shareid, it's just a resourceId. Also IMO this function should proabably  named `ExtractShareIdFromResourceId` or similar to avoid confusion.

Reply:
you're right, the naming is better